File: extensions/shared/src/main/java/app/revanced/extension/music/patches/ads/MusicAdsPatch.java
package app.revanced.extension.music.patches.ads;

import app.revanced.extension.music.settings.Settings;

@SuppressWarnings("unused")
public class MusicAdsPatch {

    public static boolean hideMusicAds() {
        return !Settings.HIDE_MUSIC_ADS.get();
    }

    public static boolean hideMusicAds(boolean original) {
        return !Settings.HIDE_MUSIC_ADS.get() && original;
    }
}

File: src/main/kotlin/app/revanced/patches/music/ads/music/MusicAdsPatch.kt
package app.revanced.patches.music.ads.music

import app.revanced.patches.shared.patch.ads.AbstractAdsPatch
import app.revanced.util.integrations.Constants.MUSIC_ADS_PATH

object MusicAdsPatch : AbstractAdsPatch(
    "$MUSIC_ADS_PATH/MusicAdsPatch;->hideMusicAds()Z"
)
